### PR TITLE
Adding services.nginx.http.server.<name> option

### PR DESCRIPTION
**Pull request by MarcWeber**
which can be used to define servers in the http section

### CODE REVIEW

**Comment by lethalman** [comment]
I wonder why isn't httpConfig enough, I'm able to add servers with httpConfig as I wish.

**Comment by edwtjo** [comment]
@lethalman isn't the added documentation worth it alone? Anyway I like it as long as I have to option to just pass a flat text config

**Comment by MarcWeber** [comment]
The reason I did it was separating port configuration from server path configuration

File: nixos/modules/services/web-servers/nginx/default.nix [hunk 1]
```diff
@@ -89,6 +156,36 @@ in
...
+        let defaultServer = optionalString server.defaultServer "default_server";
+            accessLog = optionalString (server.accessLog != "")  "error_log ${errorLog}";
+            errorLog = optionalString (server.errorLog != "")    "access_log ${accessLog}";
+            listen =
+              if server.sslCert != "" && server.sslKey != ""
+              then ''
+                listen ${server.listen} ssl ${defaultServer};
+                ssl_certificate     ${server.sslCert};
+                ssl_certificate_key ${server.sslKey};
+                ssl_protocols       SSLv3 TLSv1 TLSv1.1 TLSv1.2;
```
Comment (fpletz) [inline nixos/modules/services/web-servers/nginx/default.nix]: SSLv3 is broken due to POODLE.

Comment (MarcWeber) [inline nixos/modules/services/web-servers/nginx/default.nix]: thx, PR fixed

**Comment by zimbatm** [comment]
Closing as the powerline fonts have been added since April 2015. 76dcf48b13c27b7a2fc2373e2bf2e2e1e95ab120

**Comment by zimbatm** [comment]
Sorry wrong PR

**Comment by jagajaga** [comment]
PIng all and cc @domenkozar. LGTM

**Comment by globin** [comment]
:-1: for now, I want to review this carefully, but haven't got the time at this moment.

(We have started building something similar, but is not in a mergeable state, yet, as some things are very opinionated and I'd like to merge the efforts) 

Those interested, may want to look at the following, feel free to cherry-pick valuable commits into the branch or improve on our ideas :)
https://github.com/mayflower/nixpkgs/commit/1031c3af6af0d0a95cddee7d54b3f972d8e5bb29
https://github.com/mayflower/nixpkgs/commit/d587e789fc12e4ad2de8183c676f59c960732e0a
https://github.com/mayflower/nixpkgs/commit/d9bd2da67fb5c50d7e7f4214f5ebeef937faef67
https://github.com/mayflower/nixpkgs/commit/f89d5e4774e493c2e88dfe91213cb9241fb36444
https://github.com/mayflower/nixpkgs/commit/d7681dbba838667c280a61dcd1fe84eb1b621b00
https://github.com/mayflower/nixpkgs/commit/4de5cdfa2ed82e294c94a217a73179c4c6c4f714
https://github.com/mayflower/nixpkgs/commit/a88b7379c5457f19f450902c44c39dc24dd2de3a
https://github.com/mayflower/nixpkgs/commit/c50e243831854b3c08a80e827b8c9dd82844aea8
https://github.com/mayflower/nixpkgs/commit/0b7b77a925f3ffb95fd78d0c615d1b5c838e1202
https://github.com/mayflower/nixpkgs/commit/e3bcf876c7d39d4d8b45f6b0cc5bc050407db564
https://github.com/mayflower/nixpkgs/commit/bd77646a64df6c295ee43050a7a975d3def9bc0a
https://github.com/mayflower/nixpkgs/commit/3ce06d5877ff7f863dea552f5a03340c87abd8c9

**Comment by groxxda** [comment]
ping @globin any updates / time frame on your efforts?

**Comment by fpletz** [comment]
Here is our proposal for the nginx module: #15862

**Comment by garbas** [comment]
i'm closing this PR due to the work at #15862

**Pull request closed without merging**